container/podman/factory.go

```diff
@@ -0,0 +1,210 @@
+
+	docker "github.com/docker/docker/client"
+	"golang.org/x/net/context"
+	"k8s.io/klog/v2"
+)
+
+var ArgPodmanEndpoint = flag.String("podman", "unix:///run/podman/podman.sock", "podman endpoint")
+var ArgPodmanTLS = flag.Bool("podman-tls", false, "use TLS to connect to podman")
+var ArgPodmanCert = flag.String("podman-tls-cert", "cert.pem", "path to client certificate")
+var ArgPodmanKey = flag.String("podman-tls-key", "key.pem", "path to private key")
+var ArgPodmanCA = flag.String("podman-tls-ca", "ca.pem", "path to trusted CA")
+
+// The namespace under which podman aliases are unique.
+const PodmanNamespace = "podman"
+
+// The retry times for getting podman root dir
+const rootDirRetries = 5
+
+//The retry period for getting podman root dir, Millisecond
+const rootDirRetryPeriod time.Duration = 1000 * time.Millisecond
+
+// Regexp that identifies podman cgroups
+// --cgroup-parent have another prefix than 'libpod'
+var podmanCgroupRegexp = regexp.MustCompile(`([a-z0-9]{64})`)
+
+var podmanEnvWhitelist = flag.String("podman_env_metadata_whitelist", "", "a comma-separated list of environment variable keys matched with specified prefix that needs to be collected for podman containers")
+
+var (
+	// Basepath to all container specific information that libcontainer stores.
+	podmanRootDir string
+
+	podmanRootDirOnce sync.Once
+)
+
+func RootDir() string {
+	podmanRootDirOnce.Do(func() {
+		for i := 0; i < rootDirRetries; i++ {
+			status, err := Status()
+			if err == nil && status.RootDir != "" {
+				podmanRootDir = status.RootDir
+				break
+			} else {
+				time.Sleep(rootDirRetryPeriod)
+			}
+		}
+	})
+	return podmanRootDir
+}
+
+type podmanFactory struct {
+	machineInfoFactory info.MachineInfoFactory
+
+	client *docker.Client
+
+	// Information about the mounted cgroup subsystems.
+	cgroupSubsystems libcontainer.CgroupSubsystems
+
+	// Information about mounted filesystems.
+	fsInfo fs.FsInfo
+
+	podmanVersion []int
+
+	podmanAPIVersion []int
+
+	includedMetrics container.MetricSet
+}
+
+func (f *podmanFactory) String() string {
+	return PodmanNamespace
+}
+
+func (f *podmanFactory) NewContainerHandler(name string, inHostNamespace bool) (handler container.ContainerHandler, err error) {
+	client, err := Client()
+	if err != nil {
+		return
+	}
+
+	metadataEnvs := strings.Split(*podmanEnvWhitelist, ",")
+
+	handler, err = newPodmanContainerHandler(
+		client,
+		name,
+		f.machineInfoFactory,
+		f.fsInfo,
+		&f.cgroupSubsystems,
+		inHostNamespace,
+		metadataEnvs,
+		f.podmanVersion,
+		f.includedMetrics,
+	)
+	return
+}
+
+// Returns the Podman ID from the full container name.
+func ContainerNameToPodmanId(name string) string {
+	id := path.Base(name)
+
+	if matches := podmanCgroupRegexp.FindStringSubmatch(id); matches != nil {
+		return matches[1]
+	}
+
+	return id
+}
+
+// isContainerName returns true if the cgroup with associated name
+// corresponds to a podman container.
+func isContainerName(name string) bool {
```

Comment:
This is possitive for:
``` /system.slice/docker-72e5a5ff5eef3c4222a6551b992b9360a99122f77d2229783f0ee0946dfd800e.scope```

Is it correct behaviour? 

`CanHandleAndAccept` function comment not mention this.

Reply:
I improved the comment. Is it ok now?